esx_service/vmdk-opsd

```diff
@@ -112,32 +122,65 @@ stop() {
    if [ -n "${PID}" ]; then
       echo "Stopping $OPSD_TAG with PID=${PID}"
       kill -$SIGTERM ${PID}
+
       local COUNT=0
       while kill -0 ${PID} > /dev/null 2>&1; do
          sleep ${WAIT_KILL_SEC};
+         echo "Waiting for operations in flight to complete, please do not interrupt..."
          if [ $((COUNT++)) -gt ${NUM_RETRY} ]; then
-            echo "Sending kill -${SIGKILL} to ${PID} after kill -${SIGTERM} did not stop ${PID}"
-            kill -${SIGKILL} ${PID}
+            echo "Kill -${SIGTERM} failed to stop ${PID}"
             break
          fi
       done
    fi
 
+   cleanup_resource
+   if [ $? -ne 0 ]; then
+      exit $?
+   fi
+
+   check_running_status
```

Comment:
nit: it's also possible to use
```shell
if check_running_status # this will be true if $?==0 for check_running_status 
then 
  do this and that
fi
```
Not worth delaying the PR for the fix so just FYI

Reply:
Above change makes the code shorter, but it might cause ambiguity for new readers. Also it's consistent with existing style in this file, so I will keep it as is for now.